### PR TITLE
Click Events And Optimize Participations

### PR DESCRIPTION
**Pull request by Ap0c**
## Why are you doing this?

To support the changes in #980, we need to make sure AB test information is correctly being recorded.

Also brings the `ReferrerAcquisitionData` type into line with how it's defined in other places. It now has an `abTests: ?AcquisitionABTest[]` field instead of an `abTest: ?AcquisitionABTest` field. Dotcom still only has a single `abTest` field AFAIK (@jranks123 @Mullefa ?), so support-frontend will now translate that into an array. It should also handle any `sessionStorage` copies of `ReferrerAcquisitionData` through the same mechanism.

[**Trello Card**](https://trello.com/c/1Z71dBIk/1925-reinstate-click-events-and-add-optimize-participations-for-subs)

## Changes

- Changed `ReferrerAcquisitionData` to have an `abTests` field.
- Reinstated Ophan and GA click events for products on the subs landing pages. These events optionally take an AB test.
- Built on the work in #895 to pass Optimize experiments as AB participations to Ophan. That PR sent them through to the Payment API, this one also starts passing them through in the links to subs, using `deriveSubsAcquisitionData`.
- Prepended all Optimize tests with the string `optimize$$`.
- Updated the tests and snapshots.
- Added the featured product section to the default UK subs landing page.
- Updated the ad-free image.
- Updated the daily edition price copy.

## Screenshots

featured-section

### CODE REVIEW

**Comment by jranks123** [comment]
@Ap0c The reason for only having one in Dotcom is that the test is always associated with the component that the CTA was on, which is only ever in one test. Or at least that has always been the case so far. This is opposed to landing page, where we routinely run multiple tests at the same time

File: assets/components/featuredDigitalPack/featuredDigitalPack.jsx [hunk 1]
```diff
@@ -19,6 +23,7 @@ type PropTypes = {
   headingSize: HeadingSize,
   countryGroupId: CountryGroupId,
   url: string,
+  abTest: ComponentAbTest | void,
```
Comment (rupertbates) [inline assets/components/featuredDigitalPack/featuredDigitalPack.jsx]: wouldn't `null` be better here?

Comment (Mullefa) [inline assets/components/featuredDigitalPack/featuredDigitalPack.jsx]: flow question: why isn't this modelled as a maybe type?

File: assets/helpers/tracking/acquisitions.js [hunk 2]
```diff
@@ -167,11 +167,11 @@ const participationsToAcquisitionABTest = (participations: Participations): Acqu
...
 };
 
-// Appends all the experiment names (the keys) with '$Optimize' to avoid name
-// collision with native tests, and returns as array of AB tests.
+// Prepends all the experiment names (the keys) with 'optimize$$' to be able to
+// differentiate from native tests, and returns as array of AB tests.
 function optimizeExperimentsToAcquisitionABTest(opt: OptimizeExperiments): AcquisitionABTest[] {
   return Object.keys(opt).map(k => ({
-    name: k,
+    name: `optimize$$${k}`,
```
Comment (rupertbates) [inline assets/helpers/tracking/acquisitions.js]: The $ for the string interpolation is a bit confusing here isn't it? :-)

File: assets/components/featuredDigitalPack/featuredDigitalPack.jsx (hunk 1 above)
Comment (Ap0c) [reply assets/components/featuredDigitalPack/featuredDigitalPack.jsx]: Agree. I was using `undefined` to make the field disappear in the component event in `sendTrackingEventsOnClick`, but I think that's leaking an implementation detail of that function. We've standardised on `null` across the codebase so I'll use that.

Comment (Ap0c) [reply assets/components/featuredDigitalPack/featuredDigitalPack.jsx]: Flow's [maybe type](https://flow.org/en/docs/types/maybe/) allows three types to leak through, e.g. `?string` means `string | null | undefined`. I find that most of the time you don't actually *want* both `null` and `undefined`, you typically just want "value or absence of a value". As @rupertbates pointed out above, this should probably actually be `ComponentAbTest | null`, I'm going to change it now.

Comment (Ap0c) [reply assets/components/featuredDigitalPack/featuredDigitalPack.jsx]: Done.

Comment (Mullefa) [reply assets/components/featuredDigitalPack/featuredDigitalPack.jsx]: interesting - thanks for the info 👌

**Pull request merged**